core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -2592,11 +2601,23 @@ class KafkaApis(val requestChannel: RequestChannel,
             response.groups.addAll(results)
           }
 
+          // Remove the unauthorized topics from the member assignments and target assignments.
+          val topicsToCheck = response.groups.asScala.flatMap(_.members.asScala).flatMap { member =>
+            member.assignment.topicPartitions.asScala.map(_.topicName) ++
+              member.targetAssignment.topicPartitions.asScala.map(_.topicName)
+          }.toSet
```

Comment:
For this one, I am tempted by using a mutable Set. With it, we could basically iterate over all the groups, members and topics to populate it. It would avoid all the extract collections that we create here.